src/util/Concepts.hxx

```diff
@@ -30,12 +30,13 @@
 #pragma once
 
 #include <concepts>
+#include <util/Compiler.h>
```

Comment:
Please don't introduce this header dependency. If all you need to check is a major version, `__clang_major__` is simple enough.